### PR TITLE
feat: data - trades

### PR DESCRIPTION
**Pull request by denverbaumgartner**
# rubi-py PR

- Note - **please make sure the title of your PR follows semantic versioning** 
- Ex. fix/feat/refactor: description 
- For a breaking change, that will bump an entire version, please use !. Ex. refactor!: description 

## Description

adds in functionality for querying historical trades through the subgraph, extends the client to include this functionality. 

## What was the issue?

this data access point did not previously exist. 

## What were the changes?

updated `data/market.py`, added types for the trades query, and extended functionality in the `client`

## What type of change was this 

- [ ] fix - fixing bugs and adding small changes (X.X.X+1)
- [x] feat - introducing a new feature (X.X+1.X)
- [ ] breaking - a breaking API change (X+1.X.X)

### CODE REVIEW

File: rubi/rubi/data/market.py [hunk 1]
```diff
@@ -237,3 +253,160 @@ def get_offers(
...
+
+    def get_trades(
+        self,
+        taker: Optional[str] = None,
+        from_address: Optional[str] = None,
+        pair_name: Optional[str] = None,
+        book_side: Optional[OrderSide] = OrderSide.NEUTRAL,
+        take_gem: Optional[
+            str
+        ] = None,  # TODO: not sure we really need this given the pair_name parameter
```
Comment (adamamsmith) [inline rubi/rubi/data/market.py:265]: I would say we either go with `pair_name` or we have the `take_gem` and `buy_gem` parameters.

As this is a lover level function on the data/market.py I suggest we remove pair_name and leave that abstraction to the client.

File: rubi/rubi/client.py [hunk 2]
```diff
@@ -689,6 +689,37 @@ def get_offers(
         )
         return df
 
+    def get_trades(
+        self,
+        taker: Optional[str] = None,
+        from_address: Optional[str] = None,
+        pair_name: Optional[str] = None,
+        book_side: Optional[OrderSide] = OrderSide.NEUTRAL,
+        take_gem: Optional[str] = None,
```
Comment (adamamsmith) [inline rubi/rubi/client.py:698]: On the comment I on the `data/market.py` I think that this method is the one we abstract for the user. As such this one should have the `pair_name` but not the `take_gem` and `buy_gem` params.

File: rubi/rubi/data/market.py [hunk 3]
```diff
@@ -77,7 +91,7 @@ def get_offers(
         maker: Optional[str] = None,
         from_address: Optional[str] = None,
         pair_name: Optional[str] = None,
-        book_side: Optional[OrderSide] = None,
+        book_side: Optional[OrderSide] = OrderSide.NEUTRAL,
```
Comment (adamamsmith) [inline rubi/rubi/data/market.py:94]: If this parameter needs to be supplied it should probably not be `Optional`, instead just: 

```python
book_side: OrderSide = OrderSide.NEUTRAL,
```

File: rubi/rubi/client.py (hunk 2 above)
Comment (denverbaumgartner) [reply rubi/rubi/client.py:698]: fo sho

**Pull request merged**